pkg/operation/seed/istio/proxy_protocol_test.go

```diff
@@ -0,0 +1,216 @@
+
+		c = fake.NewFakeClientWithScheme(s)
+
+		renderer := cr.NewWithServerVersion(&version.Info{})
+		ca := kubernetes.NewChartApplier(renderer, kubernetes.NewApplier(c, meta.NewDefaultRESTMapper([]schema.GroupVersion{})))
+		Expect(ca).NotTo(BeNil(), "should return chart applier")
+
+		proxy = NewProxyProtocolGateway(deployNS, ca, chartsRootPath)
+
+		// expectedGW = &networkingv1beta1.Gateway{
+		// 	ObjectMeta: metav1.ObjectMeta{
+		// 		Labels: map[string]string{
+		// 			"app":   "istio-ingressgateway",
+		// 			"istio": "ingressgateway",
+		// 		},
+		// 	},
+		// 	Spec: v1beta1.Gateway{
+		// 		Selector: map[string]string{
+		// 			"app":   "istio-ingressgateway",
+		// 			"istio": "ingressgateway",
+		// 		},
+		// 		Servers: []*v1beta1.Server{{
+		// 			Port: &v1beta1.Port{
+		// 				Number:   uint32(8443),
+		// 				Name:     "tcp",
+		// 				Protocol: "TCP",
+		// 			},
+		// 			Hosts: []string{"*"},
+		// 		}},
+		// 	},
+		// }
+
+		// expectedVS = &networkingv1beta1.VirtualService{
+		// 	ObjectMeta: metav1.ObjectMeta{
+		// 		Labels: map[string]string{
+		// 			"app":   "istio-ingressgateway",
+		// 			"istio": "ingressgateway",
+		// 		},
+		// 	},
+		// 	Spec: v1beta1.VirtualService{
+		// 		Hosts:    []string{"blackhole.local"},
+		// 		Gateways: []string{"proxy-protocol"},
+		// 		ExportTo: []string{"."},
+		// 		Tcp: []*v1beta1.TCPRoute{
+		// 			{
+		// 				Match: []*v1beta1.L4MatchAttributes{{Port: uint32(8443)}},
+		// 				Route: []*v1beta1.RouteDestination{
+		// 					{Destination: &v1beta1.Destination{
+		// 						Host: "localhost",
+		// 						Port: &v1beta1.PortSelector{Number: 9999},
+		// 					}},
+		// 				},
+		// 			},
+		// 		},
+		// 	},
+		// }
+
+		// expectedEF = &networkingv1alpha3.EnvoyFilter{
+		// 	ObjectMeta: metav1.ObjectMeta{
+		// 		Labels: map[string]string{
+		// 			"app":   "istio-ingressgateway",
+		// 			"istio": "ingressgateway",
+		// 		},
+		// 	},
+		// 	Spec: v1alpha3.EnvoyFilter{
+		// 		WorkloadSelector: &v1alpha3.WorkloadSelector{
+		// 			Labels: map[string]string{
+		// 				"app":   "istio-ingressgateway",
+		// 				"istio": "ingressgateway",
+		// 			},
+		// 		},
+		// 		ConfigPatches: []*v1alpha3.EnvoyFilter_EnvoyConfigObjectPatch{{
+		// 			ApplyTo: v1alpha3.EnvoyFilter_LISTENER,
+		// 			Match: &v1alpha3.EnvoyFilter_EnvoyConfigObjectMatch{
+		// 				Context: v1alpha3.EnvoyFilter_GATEWAY,
+		// 				ObjectTypes: &v1alpha3.EnvoyFilter_EnvoyConfigObjectMatch_Listener{
+		// 					Listener: &v1alpha3.EnvoyFilter_ListenerMatch{
+		// 						PortNumber: uint32(8443),
+		// 						Name:       "0.0.0.0_8443",
+		// 					},
+		// 				},
+		// 			},
+		// 			Patch: &v1alpha3.EnvoyFilter_Patch{
+		// 				Operation: v1alpha3.EnvoyFilter_Patch_MERGE,
+		// 				Value: messageToStruct(&xdsAPI.Listener{
+		// 					ListenerFilters: []*listenerv2.ListenerFilter{
+		// 						{Name: "envoy.listener.proxy_protocol"},
+		// 					},
+		// 				}),
+		// 			},
+		// 		}},
+		// 	},
+		// }
```

Comment:
leftovers?

Reply:
I've spend lots of time on those tests just to realize that they would require K8S 1.18 and its not supported now. See the comment above:

```
// This entire test is commended out because istio requires K8S API 1.18
// TODO (mvladev): once we update to 1.18 enable this.
```